hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/processors/CustomMessageCallProcessor.java

```diff
@@ -118,7 +131,14 @@ public void start(@NonNull final MessageFrame frame, @NonNull final OperationTra
         // paid using gas; for example, when creating a new token. But the system contract
         // only diverts this value to the network's fee collection accounts, instead of
         // actually receiving it.
+        // We do not allow sending value to Hedera system contracts except in the case of token creation.
         if (systemContracts.containsKey(codeAddress)) {
+            if (!checkIfCreateScenario(frame)) {
```

Comment:
Nit: consider renaming `checkIfCreateScenario` to `ifTokenCreation`

Reply:
nit: Or maybe `isTokenCreate` / `isTokenCreation`, so it can be read `if is token create`, instead of `if if token create` 😜